dotCMS/src/main/java/com/dotcms/storage/FileSystemStoragePersistenceAPIImpl.java

```diff
@@ -0,0 +1,461 @@
+        return this.existsGroup(groupNameLC) && new File(this.groups.get(groupNameLC), objectPath)
+                .exists();
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param groupName {@link String} group name
+     * @return
+     */
+    @Override
+    public boolean createGroup(final String groupName) {
+        return this.createGroup(groupName, ImmutableMap.of());
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param groupName    {@link String} group name
+     * @param extraOptions {@link Map} depending on the implementation it might need extra options or not.
+     * @return
+     */
+    @Override
+    public boolean createGroup(final String groupName, final Map<String, Object> extraOptions) {
+        final String groupNameLC = groupName.toLowerCase();
+        final File rootGroup = this.groups.get(getRootGroupKey());
+        final File destBucketFile = new File(rootGroup, groupNameLC);
+        final boolean mkdirs = destBucketFile.mkdirs();
+        if(mkdirs) {
+           this.groups.put(groupNameLC, destBucketFile);
+        }
+        return mkdirs;
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param groupName {@link String} group name
+     * @return
+     */
+    @Override
+    public int deleteGroup(final String groupName) {
+        final File rootGroup = this.groups.get(getRootGroupKey());
+        final File destBucketFile = new File(rootGroup, groupName.toLowerCase());
+        if (!rootGroup.equals(destBucketFile)) {
+            final int count = countFiles(destBucketFile);
+            FileUtil.deltree(destBucketFile, true);
+            return count;
+        }
+        return 0;
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param groupName {@link String} group name
+     * @param path   {   @link String} object path
+     * @return
+     */
+    public boolean deleteObject(final String groupName, final String path) {
+        return new File(this.groups.get(groupName.toLowerCase()), path.toLowerCase()).delete();
+    }
+
+    /**
+     * {@inheritDoc}
+     * @return
+     */
+    @Override
+    public List<String> listGroups() {
+
+        return new ImmutableList.Builder<String>().addAll(this.groups.keySet()).build();
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param groupName  {@link String} the group to upload
+     * @param path       {@link String} path to upload the file
+     * @param file       {@link File}   the actual file
+     * @param extraMeta  {@link Map} optional metadata, this could be null but depending on the implementation it would need some meta info.
+     * @return
+     */
+    @Override
+    public Object pushFile(final String groupName,
+            final String path,
+            final File file,
+            final Map<String, Object> extraMeta) {
+
+        if (!this.existsGroup(groupName)) {
+
+            throw new IllegalArgumentException("The groupName: " + groupName +
+                    ", does not have any file mapped");
```

Comment:
Codacy found an issue: [The String literal ", does not have any file mapped" appears 4 times in this file; the first occurrence is on line 126](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)